crates/miden-lib/src/transaction/memory.rs

```diff
@@ -91,23 +91,30 @@ pub const OUTPUT_VAULT_ROOT_PTR: MemoryAddress = 16;
 /// The memory address at which the native account's new code commitment is stored.
 pub const NEW_CODE_ROOT_PTR: MemoryAddress = 20;
 
+/// The memory address at which the dirty flag of the storage commitment of the native account is
+/// stored.
+///
+/// This binary flag specifies whether the commitment is outdated: it holds 1 if some changes were
+/// made to the account storage since the last re-computation, and 0 otherwise.
+pub const NATIVE_ACCT_STORAGE_COMMITMENT_DIRTY_FLAG_PTR: MemoryAddress = 24;
+
 /// The memory address at which the transaction expiration block number is stored.
-pub const TX_EXPIRATION_BLOCK_NUM_PTR: MemoryAddress = 24;
+pub const TX_EXPIRATION_BLOCK_NUM_PTR: MemoryAddress = 28;
```

Comment:
Not from this PR, but why do we allocate full words for the pointers? Seems like we should be able to just use consecutive addresses - e.g.,:

```rust
pub const TX_VAULT_ROOT_PTR: MemoryAddress = 0;
pub const CURRENT_INPUT_NOTE_PTR: MemoryAddress = 1;
pub const NUM_OUTPUT_NOTES_PTR: MemoryAddress = 2;
pub const INPUT_VAULT_ROOT_PTR: MemoryAddress = 3;
pub const OUTPUT_VAULT_ROOT_PTR: MemoryAddress = 4;
pub const NEW_CODE_ROOT_PTR: MemoryAddress = 5;
pub const NATIVE_ACCT_STORAGE_COMMITMENT_DIRTY_FLAG_PTR: MemoryAddress = 7;
pub const TX_EXPIRATION_BLOCK_NUM_PTR: MemoryAddress = 7;
```

This is a very minor thing - but may be a "good first issue" to fix in the future.